core/src/json_schema.rs

```diff
@@ -282,6 +363,297 @@ pub mod simplified {
+                                 required: Some(vec!["x".into(), "y".into()]),
+                                 additional_properties: Some(false),
+                                 items: None,
+                                    any_of: None
+                                 },
+                                 Schema {
+                                     bson_type: Some(BsonType::Single(BsonTypeName::Null)),
+                                     properties: None,
+                                     required: None,
+                                     additional_properties: None,
+                                     items: None,
+                                     any_of: None
+                                }
+                            ])
+                        }.into())),
+                        any_of: None
+                    },
+                    Schema {
+                        bson_type: Some(BsonType::Single(BsonTypeName::Null)),
+                        properties: None,
+                        required: None,
+                        additional_properties: None,
+                        items: None,
+                        any_of: None
+                    }
+                ])
+            },
+            input = Schema {
+                bson_type: Some(BsonType::Multiple(vec![
+                    BsonTypeName::Array,
+                    BsonTypeName::Null
+                ])),
+                properties: None,
+                required: None,
+                additional_properties: None,
+                items: Some(Items::Single(Box::new(Schema {
+                    bson_type: Some(BsonType::Multiple(vec![
+                        BsonTypeName::Object,
+                        BsonTypeName::Null
+                    ])),
+                    properties: Some(map! {
+                    "x".into() =>
+                        Schema {
+                            bson_type: Some(BsonType::Multiple(vec![
+                                BsonTypeName::Object,
+                                BsonTypeName::Null
+                            ])),
+                            properties: Some(map! {
+                                "a".into() => Schema::default(),
+                                "b".into() => Schema::default(),
+                            }),
+                            additional_properties: Some(false),
+                            required: Some(vec!["a".into()]),
+                            items: None,
+                            any_of: None,
+                        },
+                        "y".into() => Schema::default(),
+                    }),
+                    required: Some(vec!["x".into(), "y".into()]),
+                    additional_properties: Some(false),
+                    items: None,
+                    any_of: None,
+                }))),
+                any_of: None,
+            }
+        );
+    }
+
     macro_rules! try_from_test {
```

Comment:
Did you mention that there was a specific bug you encountered that prompted this change, or is this just a refactor that improves the `try_from` code? If there was a particular bug, should we add a regression unit test for one or both of the `try_from` functions that shows it is now fixed?

Reply:
Yes, there was a multiple in a context we didn't expect (I forget exactly what) and I couldn't find a good way to handle it. Initially I thought, "well, I'll just convert to any of" but that then required called back up to the TryFrom for Schema when the context was in Atomic already (since Atomic can't handle AnyOf). At that point I decided the easiest thing to do would be to remove them all a priori, and that that would also simplify the simplification code.

Reply:
Worth adding a try_from regression test for that, or is the remove_multiple testing enough?